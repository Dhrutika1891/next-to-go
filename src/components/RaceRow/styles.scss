.table-row {
  display: grid;
  grid-template-columns: 1fr 1fr 1fr;
  gap: 20px;
  border-bottom: 1px solid grey;

  > div:nth-child(2),
  :nth-child(3) {
    justify-self: center;
  }
}
