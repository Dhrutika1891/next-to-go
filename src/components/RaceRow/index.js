import React from "react";
import classNames from "classnames";

import "./styles.scss";

function RaceRow({ meetingName, raceNumber, countDown, className = "" }) {
  return (
    <div
      className={classNames("table-row", { [`${className}`]: className })}
      data-testid="table-row"
    >
      <div data-testid="table-column-1">{meetingName}</div>
      <div data-testid="table-column-2">{raceNumber}</div>
      <div data-testid="table-column-3">{countDown}</div>
    </div>
  );
}

export default RaceRow;
