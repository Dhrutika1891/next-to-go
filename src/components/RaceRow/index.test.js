import { render } from "@testing-library/react";
import React from "react";

import RaceRow from ".";

describe("component race row", () => {
  it("should render race row", () => {
    expect.hasAssertions();

    const { getByTestId, baseElement } = render(
      <RaceRow
        meetingName="meeting name"
        raceNumber={2}
        countDown="countdown time"
      />
    );

    const raceRow = getByTestId("table-row");
    const firstRow = getByTestId("table-column-1");
    const secondRow = getByTestId("table-column-2");
    const thirdRow = getByTestId("table-column-3");

    expect(raceRow).toBeInTheDocument();
    expect(firstRow).toBeInTheDocument();
    expect(firstRow).toHaveTextContent("meeting name");
    expect(secondRow).toBeInTheDocument();
    expect(secondRow).toHaveTextContent(2);
    expect(thirdRow).toBeInTheDocument();
    expect(thirdRow).toHaveTextContent("countdown time");

    expect(baseElement).toMatchSnapshot();
  });
});
