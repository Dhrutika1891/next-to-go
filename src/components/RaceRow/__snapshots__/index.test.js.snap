// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component race row should render race row 1`] = `
<body>
  <div>
    <div
      class="table-row"
      data-testid="table-row"
    >
      <div
        data-testid="table-column-1"
      >
        meeting name
      </div>
      <div
        data-testid="table-column-2"
      >
        2
      </div>
      <div
        data-testid="table-column-3"
      >
        countdown time
      </div>
    </div>
  </div>
</body>
`;
